### ISSUE TITLE
[question] Influence of divisibility of B, C, T on 16 on Conv1d (and Conv2d perf) with CuDNN

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#55752]
CuDNN docs mention that for optimal matmul (i.e. GEMM) performance, the input matrix sizes should be divisisible by 16 (or even 256 for best tiling):
- https://docs.nvidia.com/deeplearning/performance/dl-performance-convolutional/index.html#channels
- https://docs.nvidia.com/deeplearning/performance/dl-performance-convolutional/index.html#checklist

How do these GEMM recommendations influence performance of Conv1d / Conv2d wrt spatial/temporal dimensions? Do same recommendations apply (since often convolutions under the hood are performed by GEMM on appropriate matrices)?

For convolutions we often use padding. If there're recommendations for spatial sizes, do they apply for spatial sizes before or after padding? I could not find any mention of padding (PadH, PadW) performance impact in https://docs.nvidia.com/deeplearning/performance/dl-performance-convolutional/index.html#imp-gemm-dim.

Is there a way to deduce by PyTorch profiler / NSIGHT profiler / cublas / cudnn logs deduce if convolutions / gemms are operating in the optimal mode?

It would be extremely useful if PyTorch published such recommendations (and best practices examples) in the docs (since cudnn is the dominant backend for training) or at least some discussion of this.

cc @ptrblck @ngimel 

@II245

**Comment by vadimkantorov** [comment]
It would also be nice if torch.profiler could output some hints about this, along with arithmetic intensity (if it doesn't already)

**Comment by vadimkantorov** [comment]
Alternatively, in some verbose mode even the regular Linear/Conv ops could print warning messages about problems with divisibility
